layers/amqp/Dockerfile

```diff
@@ -19,23 +20,16 @@ RUN set -xe; \
     cmake --build . --target install
 
 RUN if [[ "$PHP_VERSION" = "80" ]] ; then pecl install amqp-1.11.0beta ; else pecl install amqp ; fi
-RUN cp /opt/bref/lib/php/extensions/no-debug-non-zts-*/amqp.so /tmp/amqp.so
-RUN echo 'extension=/opt/bref-extra/amqp.so' > /tmp/ext.ini
+RUN cp `php-config --extension-dir`/amqp.so /tmp/amqp.so
+RUN echo 'extension=amqp.so' > /tmp/ext.ini
+
+RUN php /bref/lib-copy/copy-dependencies.php /tmp/amqp.so /tmp/extension-libs
+
 
 # Build the final image with just the files we need
 FROM scratch
 
 # Copy things we installed to the final image
-COPY --from=ext /opt/bref/include/amqp.h /opt/bref/include/amqp.h
-COPY --from=ext /opt/bref/include/amqp_framing.h /opt/bref/include/amqp_framing.h
-COPY --from=ext /opt/bref/include/amqp_ssl_socket.h /opt/bref/include/amqp_ssl_socket.h
-COPY --from=ext /opt/bref/include/amqp_tcp_socket.h /opt/bref/include/amqp_tcp_socket.h
-
-COPY --from=ext /opt/bref/lib64/librabbitmq.a /opt/bref/lib64/librabbitmq.a
-COPY --from=ext /opt/bref/lib64/librabbitmq.so /opt/bref/lib64/librabbitmq.so
-COPY --from=ext /opt/bref/lib64/librabbitmq.so.4 /opt/bref/lib64/librabbitmq.so.4
-COPY --from=ext /opt/bref/lib64/librabbitmq.so.4.3.0 /opt/bref/lib64/librabbitmq.so.4.3.0
-COPY --from=ext /opt/bref/lib64/pkgconfig/librabbitmq.pc /opt/bref/lib64/pkgconfig/librabbitmq.pc
-
-COPY --from=ext /tmp/amqp.so /opt/bref-extra/amqp.so
```

Comment:
Interesting. Can you elaborate on why it is save to remove these?

Reply:
https://github.com/brefphp/aws-lambda-layers/blob/main/utils/lib-copy/copy-dependencies.php

it seams scary =)

Should we just trust the tests?

Reply:
Yeah that's the new script that automates copying the system dependencies used by a binary/library. The goal is to make all the Dockerfiles much simpler, easier to write, and more robust, so that we don't have to manually identify and pick the correct files.

It is definitely scary 😅 Any regression we can fix and add more tests to cover, but yeah, we are still at the risk of regressions for unexpected reasons.

All the `.so` files should be correctly copied by the script (that's what the script focuses on right now).

I don't know why some `.h`, .`pc` and `.a` files were copied in the exported layers. I know `.so` are libraries and we need them, but I don't know if those other files are actually needed at runtime.

Any idea? Maybe @shouze has an idea about these files?

Reply:
for `.h` and `.pc` files I see no reasons as they're files only useful for compilation/linking use cases.

`.a` files are static version of libraries so effectively I see no better reason to copy them because they should be linked inside a binary to work.

`.so` which are dynamic libraries should be enough.

Reply:
Awesome, thanks for confirming!

Reply:
Let's try this then =)